perfkitbenchmarker/virtual_machine.py

```diff
@@ -285,36 +314,65 @@ def RemoteCopy(self, file_path, remote_path='', copy_to=True):
                     (retcode, full_cmd, stdout, stderr))
       raise errors.VmUtil.SshConnectionError(error_text)
 
+  def _PushLongRunningCommandScripts(self):
+    """Pushes the scripts required by LongRunningRemoteCommand to this VM.
+
+    If the scripts have already been placed on the VM, this is a noop.
+    """
+    with self._has_remote_command_script_lock:
+      if not self._has_remote_command_script:
+        for f in (EXECUTE_COMMAND, WAIT_FOR_COMMAND):
+          self.PushDataFile(f, os.path.join(vm_util.VM_TMP_DIR,
+                                            os.path.basename(f)))
+
   def LongRunningRemoteCommand(self, command):
     """Runs a long running command on the VM in a robust way.
 
+    Executes a long running command via a pair of scripts on the VM:
```

Comment:
Good point. Maybe a name change would make this clearer. AsyncRemoteCommand?

You're right that it should work for all commands independent of execution duration. The overhead is:

1. A second SSH connection.
1. stdout/stderr being buffered to disk rather than streamed to the client.
1. More data in the logs (as implemented).

I think the first and third are good reasons to leave most of the code as-is, but agree that we should move more code to this method.